src/main/java/uk/ac/ebi/ampt2d/commons/accession/core/BasicAccessioningService.java

```diff
@@ -147,9 +148,28 @@ public BasicAccessioningService(AccessionGenerator<MODEL, ACCESSION> accessionGe
     }
 
     @Override
-    public List<AccessionWrapper<MODEL, HASH, ACCESSION>> getByAccessions(List<ACCESSION> accessions,
-                                                                          boolean hideDeprecated) {
-        return dbService.findAllAccessionMappingsByAccessions(accessions);
+    public List<AccessionWrapper<MODEL, HASH, ACCESSION>> getByAccessionIds(List<ACCESSION> accessions,
+                                                                            boolean hideDeprecated) {
+        final List<AccessionWrapper<MODEL, HASH, ACCESSION>> allAccessionMappingsByAccessions
+                = dbService.findAllAccessionMappingsByAccessions(accessions);
+        Map<ACCESSION, Integer> accessionToMaxVersion = generateAccessionToMaxVersion(allAccessionMappingsByAccessions);
```

Comment:
can the function name be getMapOfAccessionToMaxVersion instead generateAccessionToMaxVersion ?

Reply:
It is not a getter, processes the current information and generates the map.

Reply:
ahh sorry in that case generateMapOfAccessionToMaxVersion ?

Reply:
No, you already know it is a map from the return. Adding that information to the method name is a little bit redundant.